wire-ios/Wire-iOS/Sources/UserInterface/Settings/ChangingDetails/ChangeEmailViewModel.swift

```diff
@@ -0,0 +1,52 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import WireSyncEngine
+
+final class ChangeEmailViewModel {
```

Comment:
Have you considered making this object observable? That way if we migrate the view to SwiftUI we would likely only need to change the view, and the view model could be used as is.

Reply:
Observable is available from iOS 17 and I don't want to have two implementations for the VM. We can consider it at a later stage. 

```'Observable()' is only available in iOS 17.0 or newer```

Reply:
Sorry I should have been more specific. For this I meant [ObservableObject](https://developer.apple.com/documentation/combine/observableobject) which is available from iOS 13. But anyway I feel free to discard this comment because it would require larger changes outside the scope of your PR.